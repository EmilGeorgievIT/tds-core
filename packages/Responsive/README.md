# TDS Core: Responsive

Respond to device features, most commonly the browser viewport size.
