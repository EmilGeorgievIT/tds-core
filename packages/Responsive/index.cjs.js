const Responsive = require('./dist/index.cjs')

module.exports = Responsive
