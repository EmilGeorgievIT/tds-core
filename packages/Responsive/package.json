{
  "name": "@tds/core-responsive",
  "version": "1.0.0",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "license": "MIT",
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  },
  "dependencies": {
    "prop-types": "^15.5.10"
  },
  "devDependencies": {
    "react-media": "^1.8.0"
  }
}
