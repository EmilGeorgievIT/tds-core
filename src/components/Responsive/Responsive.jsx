import React from 'react'
import PropTypes from 'prop-types'
// import MediaQuery from 'react-responsive'

import safeRest from '../../utils/safeRest'

// import styles from './Responsive.modules.scss'

// const breakpoints = {
//   small: 576,
//   medium: 768,
//   large: 992,
//   xl: 1200,
// }

// const Small = props => <MediaQuery {...props} minWidth={breakpoints.small} />
// const Medium = props => <MediaQuery {...props} minWidth={breakpoints.medium} />
// const Large = props => <MediaQuery {...props} minWidth={breakpoints.large} />
// const XL = props => <MediaQuery {...props} minWidth={breakpoints.xl} />

const Responsive = ({ children, ...rest }) => <div {...safeRest(rest)}>{children}</div>

Responsive.propTypes = {
  query: PropTypes.string,
  children: PropTypes.node.isRequired,
}

Responsive.defaultProps = {
  query: undefined,
}

export default Responsive
