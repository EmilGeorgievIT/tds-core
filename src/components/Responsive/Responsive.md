The Responsive component is a thin wrapper over the [**react-media**](https://github.com/ReactTraining/react-media) community 
component. We encourage mobile-first development by using the `minWidth` prop exclusively, though you're welcome to use 
`maxWidth` or any of the other props supplied by **react-media** if the need arises; **note** if you do use props 
supplied by react-media, you must provide pixel values and not breakpoint keys as used by the Responsive component.

### Breakpoints

Using the `minWidth` and `maxWidth` props, you can pass a breakpoint key. They represent the following:

| Breakpoint key | Value as `minWidth` | Value as `maxWidth` |
| -------------- | ------------------- | ------------------- |
| `sm` | 576px | 575px |
| `md` | 768px | 767px |
| `lg` | 992px | 991px |
| `xl` | 1200px | 1199px |

### Basic usage

```jsx
<div>
  <Responsive minWidth="sm" maxWidth="md">
    <Heading level="h1">Hello I'm a heading in XS and SM</Heading>
  </Responsive>
  <Responsive minWidth="md" maxWidth="lg">
    <Card>
      Hello I'm a Card in MD
    </Card>
  </Responsive>
  <Responsive minWidth="lg">
    <Button>Hello I'm a Button in L and XL</Button>
  </Responsive>
</div>
```

### Passing a child function

Since the Responsive component wraps **react-media**, you may pass a function as the only child to perform logical 
operations based on whether or not the query matches the browser viewport.

```jsx
<Responsive minWidth="sm">
{
  (matches) => 
    matches ? "You're using a large display." : "You're viewing this on mobile."
}
</Responsive>
```
