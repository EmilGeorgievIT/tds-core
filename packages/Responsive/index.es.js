import Responsive from './dist/index.es'

export default Responsive
