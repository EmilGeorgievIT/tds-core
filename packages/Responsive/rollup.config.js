import configure from '../../config/rollup.config'
import { dependencies } from './package.json'

export default configure({
  input: './Responsive.jsx',
  css: false,
  dependencies,
})
