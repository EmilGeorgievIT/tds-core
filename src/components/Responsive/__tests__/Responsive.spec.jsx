import React from 'react'
import { shallow } from 'enzyme'

import Responsive from '../Responsive'

describe('Responsive', () => {
  const doShallow = (props = {}) => shallow(<Responsive {...props}>This is children</Responsive>)

  it('renders', () => {
    const responsive = doShallow()

    expect(responsive).toMatchSnapshot()
  })

  it('does other things', () => {
    const responsive = doShallow()

    expect(responsive).toBePresent()
  })

  it('passes additional attributes to the element', () => {
    const responsive = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(responsive).toHaveProp('id', 'the-id')
    expect(responsive).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const responsive = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(responsive).not.toHaveProp('className', 'my-custom-class')
    expect(responsive).not.toHaveProp('style')
  })
})
