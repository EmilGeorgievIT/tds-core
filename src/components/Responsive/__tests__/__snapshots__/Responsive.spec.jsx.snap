// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responsive renders 1`] = `
<div>
  This is children
</div>
`;
